src/tailwindcss.worker.ts

```diff
@@ -1,174 +1,3 @@
-
-  const state: JitState = {
-    version: '3.0.0',
-    config,
-    enabled: true,
-    modules: {
-      postcss: { module: postcss, version: '' },
-      postcssSelectorParser: { module: postcssSelectorParser },
-      jit: {
-        createContext: { module: createContext },
-        expandApplyAtRules: { module: expandApplyAtRules },
-        generateRules: { module: generateRules },
-      },
-    },
-    classNames: {
-      classNames: {},
-      context: {},
-    },
-    // ClassNames: extractClasses(lspRoot),
-
-    jit: true,
-    jitContext,
-    separator: ':',
-    screens: isObject(config.theme.screens) ? Object.keys(config.theme.screens) : [],
-    editor: {
-      userLanguages: {},
-      // @ts-expect-error this is poorly typed
-      // eslint-disable-next-line require-await
-      async getConfiguration() {
-        return {
-          editor: { tabSize: 2 },
-          tailwindCSS: {
-            validate: true,
-            classAttributes: ['class', 'className'],
-            lint: {
-              cssConflict: 'warning',
-              invalidApply: 'error',
-              invalidScreen: 'error',
-              invalidVariant: 'error',
-              invalidConfigPath: 'error',
-              invalidTailwindDirective: 'error',
-              recommendedVariantOrder: 'warning',
-            },
-          },
-        };
-      },
-    },
-  };
-
-  state.variants = getVariants(state);
-
-  state.classList = jitContext
-    .getClassList()
-    .filter((className) => className !== '*')
-    .map((className) => [className, { color: getColor(state, className) }]);
-
-  const getTextDocument = (uri: string, languageId: string): TextDocument | undefined => {
-    const models = ctx.getMirrorModels();
-    for (const model of models) {
-      if (String(model.uri) === uri) {
-        return TextDocument.create(uri, languageId, model.version, model.getValue());
-      }
-    }
-  };
-
-  return {
-    doComplete(uri, languageId, position, context) {
-      const textDocument = getTextDocument(uri, languageId);
-
-      if (!textDocument) {
-        return;
-      }
-
-      return doComplete(state, textDocument, position, context);
-    },
-
-    doHover(uri, languageId, position) {
-      const textDocument = getTextDocument(uri, languageId);
-
-      if (!textDocument) {
-        return;
-      }
-
-      return doHover(state, textDocument, position);
-    },
-
-    doValidate(uri, languageId) {
-      const textDocument = getTextDocument(uri, languageId);
-
-      if (!textDocument) {
-        return [];
-      }
-
-      return doValidate(state, textDocument);
-    },
-
-    getDocumentColors(uri, languageId) {
-      const textDocument = getTextDocument(uri, languageId);
-
-      if (!textDocument) {
-        return [];
-      }
-
-      return getDocumentColors(state, textDocument);
-    },
-
-    resolveCompletionItem(item) {
-      return resolveCompletionItem(state, item);
-    },
-  };
-});
+initialize();
```

Comment:
I think we can keep the logic inside this file, call `initialize()`, and allow the user to call `initialize()` as well. All `initialize()` does is register `self.onmessage` (https://github.com/remcohaszing/monaco-worker-manager/blob/main/worker.ts#L27), so calling it twice shouldn’t be a problem.

I think it’s simpler for us to manage having fewer output files and for the user to not have to make this distinction. Everything related to the worker, whether using directly or customizing, happens in `monaco-tailwindcss/tailwindcss.worker.js`.

Reply:
yes you're right! I was at first also searching for a a way to not have two files... But sadly there is (and cannot) be a python __name__ == '__main__'

And self.onmessage will just be overridden this is fine. (unless there will be other logic implemented in the worker helper but you would remember then this 'feature' ^^